### PR TITLE
exclude ignored messages (maybe Closes #1472)

### PR DESCRIPTION
**Pull request by SvetlanaTkachenko**
untested

### CODE REVIEW

**Comment by digitalcircuit** [comment]
I'd try to test this, but I'm not sure how the `Away Log` feature works in the first place.  Do you have any suggestions?

**Comment by SvetlanaTkachenko** [comment]
When detached, highlights are added to away log.

To repro this try to

1. ignore someone (your alt nick if you want)
2. detach quassel client
3. have the someone highlight you
4. attach back

expected result - their message is not in the away log because they are ignored
actual result - their message is in the away log still

File: src/qtui/awaylogfilter.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@
  *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.         *
  ***************************************************************************/
 
+#include "clientignorelistmanager.h"
```
Comment (digitalcircuit) [inline src/qtui/awaylogfilter.cpp]: **Minor:** This should go below `#include "awaylogfilter.h"`, to keep includes in alphabetical order.

File: src/qtui/awaylogfilter.cpp [hunk 2]
```diff
@@ -33,16 +34,25 @@ bool AwayLogFilter::filterAcceptsRow(int sourceRow, const QModelIndex &sourcePar
...
 
+    // only show messages that were sent after the user detached
     if (Client::networkModel()->lastSeenMsgId(bufferId) >= sourceModel()->data(source_index, MessageModel::MsgIdRole).value<MsgId>())
         return false;
+    
+    // ignorelist handling
+    // only match if message is not flagged as server msg
+    if (!(flags & Message::ServerMsg) && Client::ignoreListManager()
+        && Client::ignoreListManager()->match(source_index.data(MessageModel::MessageRole).value<Message>(), Client::networkModel()->networkName(bufferId)))
+return false;
```
Comment (digitalcircuit) [inline src/qtui/awaylogfilter.cpp]: **Minor:** `return false` should be indented to be within the `if` statement scope.

```cpp
    [...]
    if (!(flags & Message::ServerMsg) && Client::ignoreListManager()
            && Client::ignoreListManager()->match(source_index.data(MessageModel::MessageRole).value<Message>(), Client::networkModel()->networkName(bufferId)))
        return false;
    [...]
```

**Pull request merged**